kratos/geometries/triangle_3d_3.h

```diff
@@ -1712,7 +1646,89 @@ template<class TPointType> class Triangle3D3
     ///@}
     ///@name Private Operations
     ///@{
+    
+    /**
+     * Returns the local coordinates of a given arbitrary point 
+     * @param rResult The vector containing the local coordinates of the point
+     * @param rPoint The point in global coordinates
+     * @param IsInside The flag that checks if we are computing IsInside (is common for seach to have the nodes outside the geometry)
+     * @return The vector containing the local coordinates of the point
+     */
+    CoordinatesArrayType& PointLocalCoordinatesImplementation(
+        CoordinatesArrayType& rResult,
+        const CoordinatesArrayType& rPoint,
+        const bool IsInside = false
+        )
+    {
+        bounded_matrix<double,3,3> X;
+        bounded_matrix<double,3,2> DN;
+        for(unsigned int i=0; i<this->size();i++)
+        {
+            X(0,i ) = this->GetPoint( i ).X();
+            X(1,i ) = this->GetPoint( i ).Y();
+            X(2,i ) = this->GetPoint( i ).Z();
+        }
+
+        double tol = 1.0e-8;
+        int maxiter = 1000;
+
+        Matrix J = ZeroMatrix( 2, 2 );

```

Comment:
J and invJ could be bounded matrices in this function

Reply:
True